src/components/ContentSidebar/ActivityFeed/task/Task.js

```diff
@@ -126,7 +126,8 @@ class Task extends React.Component<Props> {
                                                     shouldShowActions={
                                                         onAssignmentUpdate !== noop &&
                                                           currentUser &&
-                                                          assigned_to.id === currentUser.id
+                                                          assigned_to.id === currentUser.id &&
```

Comment:
Let's move this to keep the UI there but blocking the action from doing anything. This will help avoid flashing in the UI